File: atcoder/abc441/b.py
N, M = map(int, input().split())
S = set(input())
T = set(input())
Q = int(input())

ans = []
for _ in range(Q):
    w = input()

    all_in_s = all((c in S) for c in w)
    any_not_in_s = any((c not in S) for c in w)
    all_in_t = all((c in T) for c in w)
    any_not_in_t = any((c not in T) for c in w)

    if all_in_s and any_not_in_t:
        ans.append("Takahashi")
    elif all_in_t and any_not_in_s:
        ans.append("Aoki")
    else:
        ans.append("Unknown")

print("\n".join(ans))

File: atcoder/abc441/b.rs
#![allow(unused)]

use std::collections::HashSet;

use proconio::{input, marker::Chars};

fn main() {
    input! {
        n: usize,
        m: usize,
        s: Chars,
        t: Chars,
        q: usize,
        asks: [Chars; q],
    }

    let s = HashSet::<char>::from_iter(s.iter().cloned());
    let t = HashSet::<char>::from_iter(t.iter().cloned());

    let mut ans = vec![];
    for w in asks {
        let all_in_s = w.iter().all(|c| s.contains(c));
        let all_in_t = w.iter().all(|c| t.contains(c));

        if all_in_s && !all_in_t {
            ans.push("Takahashi");
        } else if all_in_t && !all_in_s {
            ans.push("Aoki");
        } else {
            ans.push("Unknown")
        }
    }

    println!("{}", join(&ans, "\n"));
}

fn join<T: ToString>(arr: &[T], sep: &str) -> String {
    arr.iter()
        .map(|x| x.to_string())
        .collect::<Vec<String>>()
        .join(sep)
}
